### ISSUE TITLE
Dashboard access  Failure "code": 503 -Metric client health check failed connection refused 

### ISSUE THREAD

**Issue by abhassine** [kubernetes/dashboard#5533]
Started to use your following helm chart version 2.0.5 to install kube dashboard and faced this issue when I’m trying to connect :
 
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
   
  },
  "status": "Failure",
  "message": "no endpoints available for service \"https:kubernetes-dashboard:\"",
  "reason": "ServiceUnavailable",
  "code": 503
}
 
 
And the guilty is this following error I’m getting within kubernetes-dashboard container

Metric client health check failed: Get "http://127.0.0.1:8000/healthz?timeout=32s": dial tcp 127.0.0.1:8000: connect: connection refused. Retrying in 30 seconds.
 
 
However components are running fine :
 
kubernetes-dashboard   kubernetes-dashboard-687845b445-hsnvq                             2/2       Running   0          5h9m
 
 
Thank you in advance for your help

##### Environment

```
Installation method: HELM v3 Terraform provider
Kubernetes version: EKS 1.17
Dashboard version: 2.0.3 
Operating system: Amazon Linux 
Node.js version ('node --version' output): N/A
Go version ('go version' output): N/A
```

##### Steps to reproduce

We deploying using Terraform HELM provider latest version with Terraform Enterprise.

provider "helm" {
version = "1.3.0"
kubernetes {
host = module.eks_cluster.eks_endpoint
cluster_ca_certificate = base64decode(module.eks_cluster.eks_certificate_authority)
token = data.aws_eks_cluster_auth.cluster_auth.token #module.eks_cluster.auth_token
load_config_file = false
}
}

##### Observed result

connect :

I tryed to connect Dashboard like following

$ kubectl proxy

and access Dashboard at:

http://localhost:/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/.

and I get following error 
 
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
   
  },
  "status": "Failure",
  "message": "no endpoints available for service \"https:kubernetes-dashboard:\"",
  "reason": "ServiceUnavailable",
  "code": 503
}
##### Expected result

Connect to the dashboard

##### Comments

Dashboard Working and accessible when deploying with Kustomize

**Comment by floreks** [comment]
Metric client health check has nothing to do with being able to access it. It does not influence Dashboard in any way other than showing metrics.

It has to be a configuration issue on your side. There are no steps to reproduce, no information about how you deployed it and how are you trying to access it.

Fill out the form properly.

**Comment by abhassine** [comment]
Thanks for your response.
I know for Metrics however I just put all the issue I noticed

Started to use your following helm chart version 2.0.5 to install kube dashboard and faced this issue when I’m trying to connect :

{
"kind": "Status",
"apiVersion": "v1",
"metadata": {

},
"status": "Failure",
"message": "no endpoints available for service "https:kubernetes-dashboard:"",
"reason": "ServiceUnavailable",
"code": 503
}

And the guilty is this following error I’m getting within kubernetes-dashboard container

Metric client health check failed: Get "http://127.0.0.1:8000/healthz?timeout=32s": dial tcp 127.0.0.1:8000: connect: connection refused. Retrying in 30 seconds.

However components are running fine :

kubernetes-dashboard kubernetes-dashboard-687845b445-hsnvq 2/2 Running 0 5h9m

Thank you in advance for your help

Environment

Installation method: HELM v3 Terraform provider
Kubernetes version: EKS 1.17
Dashboard version: 2.0.3 
Operating system: Amazon Linux 
Node.js version ('node --version' output): N/A
Go version ('go version' output): N/A
Steps to reproduce

We deploying using Terraform HELM provider latest version with Terraform Enterprise.

provider "helm" {
  version                  = "1.3.0"
  kubernetes {
    host                   = module.eks_cluster.eks_endpoint
    cluster_ca_certificate = base64decode(module.eks_cluster.eks_certificate_authority)
    token                  = data.aws_eks_cluster_auth.cluster_auth.token #module.eks_cluster.auth_token
    load_config_file       = false
  }
}


Observed result

connect :

I tryed to connect Dashboard like following 

$ kubectl proxy

and access Dashboard at:

http://localhost:<port>/api/v1/namespaces/kubernetes-dashboard/services/https:kubernetes-dashboard:/proxy/.

Expected result

Connect to the dashboard

Comments

Dashboard Working and accessible when deploying with Kustomize
